src/main/java/org/jenkinsci/test/acceptance/po/Job.java

```diff
@@ -408,4 +408,15 @@ public ScmPolling pollScm() {
 
         return links;
     }
+
+    /**
+     * Deletes the current job
+     */
+    public void delete() {
+        this.open();
+        clickLink("Delete Project");
```

Comment:
It is responsibility of given job page object to know that - I see no reason to make the test responsible for choosing the correct label. I am fine leaving it as it is and override/rework when needed.